packages/utils/src/instrument/handlers.ts

```diff
@@ -37,8 +37,12 @@ export function resetInstrumentationHandlers(): void {
 /** Maybe run an instrumentation function, unless it was already called. */
 export function maybeInstrument(type: InstrumentHandlerType, instrumentFn: () => void): void {
   if (!instrumented[type]) {
-    instrumentFn();
-    instrumented[type] = true;
+    try {
+      instrumentFn();
+      instrumented[type] = true;
```

Comment:
I wonder if we should always set `instrumented[type] = true`

Reply:
good point! Maybe it is safer, otherwise we may double-patch parts of the code (if e.g. some stuff was wrapped but other wasn't)? 🤔

Reply:
I guess it is safe to just set it if the instrumentation did not throw? How would we be double patching it in this case? 🤔

Reply:
`instrumentFn` may do something like this:

```js
function instrumentFn() {
  monkeyPatchFn1();
  monkeyPatchFn2();
  monkeyPathFn3();
}
```

if it throws in `monkeyPatchFn3`, fn1 and fn2 would already be patched. So if we would not set the flag in this case, next time we'd run `instrumentFn` again, and patch fn1 and fn2 again (and likely, it would fail again at fn3 anyhow).

Reply:
Ah ok I didn't think of such a case 👍